### ISSUE TITLE
Regression for scala `3.0.x` at `3.2.12` release.

### ISSUE THREAD

**Issue by catap** [scalatest/scalatest#2123]
Unfortunately `3.1.x` isn't binary compatible for `3.0.x` [details](https://www.scala-lang.org/blog/2021/10/21/scala-3.1.0-released.html) and an easy way since `3.1.2` is released is using `ThisBuild / scalaOutputVersion` or similar things as [suggested here](https://www.scala-lang.org/blog/2022/04/12/scala-3.1.2-released.html).

Right now a released version of `3.2.12` is broken for `3.0.x` scala. An attempt to use it leads to error:
```
[error] error while loading $throws$package$,
[error] class file scala/runtime/$throws$package.class is broken, reading aborted with class dotty.tools.tasty.UnpickleException
[error] TASTy signature has wrong version.
[error]  expected: {majorVersion: 28, minorVersion: 0}
[error]  found   : {majorVersion: 28, minorVersion: 1}
[error] 
[error] This TASTy file was produced by a more recent, forwards incompatible release.
[error] To read this TASTy file, please upgrade your tooling.
[error] The TASTy file was produced by Scala 3.1.0-bin-nonbootstrapped.
```

**Comment by cheeseng** [comment]
@catap While I am not really against supporting both Scala 3.0 and 3.1, which as first thought will be a nice thing to do at least for library maintainer, I tend to agree with @sjrd 's argument to encourage the users to move up to Scala 3.1 as we discussed previously for scala-js here (which also uses forward compatibility): 

https://github.com/scalatest/scalatest/issues/1830#issuecomment-651922821

From Scala 3 end user point of view, I don't see migrating from 3.0 to 3.1 really a challenging move, or there's a serious problem with 3.1 as compared to 3.0.

Anyway, what's out with 3.2.12 is out, I am open about the dottyOutputVersion in 3.2.13, @bvenners what do you think?

**Comment by cheeseng** [comment]
@catap One more thing is that the scala-xml 2.1.0 that we depends on is also using Scala 3.1 and not using dottyOutputVersion also, I wonder if that will become an issue if we use dottyOutputVersion.

**Comment by catap** [comment]
@cheeseng because it is a test library I strongly support an idea to have 3.0 :)

Anyway, rationally I've checked open / closed regression inside dotty and hasn't got anything worth to support it.

To be honest, open an issue (this one) and make it clear that 3.0 isn't supported anymore fine for me.

**Comment by cheeseng** [comment]
> @cheeseng because it is a test library I strongly support an idea to have 3.0 :)
> 
> Anyway, rationally I've checked open / closed regression inside dotty and hasn't got anything worth to support it.
> 
> To be honest, open an issue (this one) and make it clear that 3.0 isn't supported anymore fine for me.
> 
> From another hand you're supporting scala before 2.12... :)

Yes we're still supporting Scala 2.10-2.13 because upgrading from 2.13 to 3.0 isn't as straight forward, the effort upgrade a normal size project building with 2.12 to 2.13 shall easily suppress effort to upgrade from 3.0 to 3.1.  Scala 3 changed the binary-compatible scheme as compared to Scala 2.x, if Scala 3 scheme was unchanged we'll definitely have to support build for Scala 3.0 and 3.1, just like we have been doing with Scala 2.x.  :)

**Comment by catap** [comment]
@cheeseng from another hand `3.2` is also different language: https://github.com/lampepfl/dotty/commit/71f5da349ad3cd9fdb3a6b69d4e954dc34764144

And `3.0` is still default version of language.

**Comment by catap** [comment]
Just for a records: scala-native also blocks such feature: https://github.com/scala-native/scala-native/issues/2606
